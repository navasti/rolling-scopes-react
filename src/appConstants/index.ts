import { FormFields } from 'types';
import * as yup from 'yup';

export enum Limits {
  pokemon = 20,
  type = 10,
  move = 15,
}

export enum ErrorStatuses {
  success = '2',
  client = '4',
  server = '5',
}

export enum AvailableTabs {
  pokemons = 'pokemons',
  moves = 'moves',
  types = 'types',
}

export enum Fields {
  birthday = 'birthday',
  consent = 'consent',
  gender = 'gender',
  avatar = 'avatar',
  shiny = 'shiny',
  name = 'name',
  type = 'type',
}

export enum ErrorMessages {
  name = 'Name must be at least 2 characters long.',
  consent = 'Regulations consent is required.',
  birthday = 'Birthday is required.',
  type = 'Main type is required.',
  gender = 'Gender is required.',
}

export enum Types {
  fire = 'fire',
  grass = 'grass',
  poison = 'poison',
  flying = 'flying',
  ground = 'ground',
  psycho = 'psycho',
  water = 'water',
  ice = 'ice',
}

export const NONE = 'none';
export const MALE = 'male';
export const FEMALE = 'female';
export const INPUT_VALUE_KEY = 'input-value';
export const SUCCESS_MESSAGE = 'Pokemon added successfuly';
export const LINKS = ['home', 'about', 'form', 'invalid', 'details'] as const;
export const SEARCH_BAR_INSTRUCTIONS = [
  'Type and press enter to search for specific pokemon, type or move.',
  'Clear input and press enter to search for all pokemons, types and moves.',
] as const;

export const TABS = Object.values(AvailableTabs);
export const POKEMON_TYPES = Object.values(Types);

export const FORM_VALIDATION_SCHEMA = yup.object().shape({
  type: yup.string().required(ErrorMessages.type),
  name: yup.string().min(2, ErrorMessages.name).required(ErrorMessages.name),
  gender: yup.string().nullable().required(ErrorMessages.gender),
  avatar: yup
    .mixed()
    .test('fileSize', 'The file is too large', (value) =>
      !value?.length ? true : value?.[0]?.size <= 200000
    ),
  birthday: yup.string().required(ErrorMessages.birthday),
  consent: yup.bool().oneOf([true], ErrorMessages.consent),
  shiny: yup.bool(),
});

const BASE = 'https://pokeapi.co/api/v2';

export const API = {
  POKEMON_LIMIT: `?limit=${Limits.pokemon}`,
  TYPE_LIMIT: `?limit=${Limits.type}`,
  MOVE_LIMIT: `?limit=${Limits.move}`,
  POKEMON: `${BASE}/pokemon`,
  TYPE: `${BASE}/type`,
  MOVE: `${BASE}/move`,
  BASE,
};

export const FIELDS_VALIDATION_BY_NAME = {
  TEXT: [Fields.name, Fields.birthday, Fields.type],
  CHECK: [Fields.consent, Fields.gender],
};

export const DEFAULT_VALUES: FormFields = {
  consent: false,
  avatar: null,
  gender: MALE,
  shiny: false,
  birthday: '',
  name: '',
  type: '',
};
