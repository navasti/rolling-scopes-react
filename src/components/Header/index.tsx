import { LocationBadge } from './components';
import { Link } from 'react-router-dom';
import * as Styled from './styled';
import React from 'react';

interface Props {
  componentName: string;
  location: string;
}

export class Header extends React.Component<Props, unknown> {
  constructor(props: Props) {
    super(props);
  }
  render() {
    return (
      <Styled.Header>
        <Styled.Nav role="navigation">
          <Styled.LinkWrapper>
            <Link role="link" to="/">
              Home
            </Link>
          </Styled.LinkWrapper>
          <Styled.LinkWrapper>
            <Link role="link" to="/about">
              About
            </Link>
          </Styled.LinkWrapper>
          <Styled.LinkWrapper>
            <Link role="link" to="/invalid">
              Invalid
            </Link>
          </Styled.LinkWrapper>
        </Styled.Nav>
        <LocationBadge componentName={this.props.componentName} location={this.props.location} />
      </Styled.Header>
    );
  }
}
