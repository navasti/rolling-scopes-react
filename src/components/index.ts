export { Loader } from './Loader';
export { Modal } from './Modal';
