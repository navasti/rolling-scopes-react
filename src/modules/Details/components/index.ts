export { MoveCardDetails } from './MoveCardDetails';
export { TypeCardDetails } from './TypeCardDetails';
export { PokemonCardDetails } from './PokemonCardDetails';
