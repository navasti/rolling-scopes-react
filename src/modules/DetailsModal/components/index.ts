export { TypeModalContent } from './TypeModalContent';
export { MoveModalContent } from './MoveModalContent';
export { PokemonModalContent } from './PokemonModalContent';
