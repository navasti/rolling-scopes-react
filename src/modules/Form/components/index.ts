export { BirthdayField } from './BirthdateField';
export { ErrorMessage } from './ErrorMessage';
export { ConsentField } from './ConsentField';
export { AvatarField } from './AvatarField';
export { GenderField } from './GenderField';
export { ShinyField } from './ShinyField';
export { TypeField } from './TypeField';
export { NameField } from './NameField';
export { FormCard } from './FormCard';
