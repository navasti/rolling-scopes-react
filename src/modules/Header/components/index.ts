export { LocationBadge } from './LocationBadge';
export { Navigation } from './Navigation';
