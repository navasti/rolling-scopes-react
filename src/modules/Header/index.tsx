import { LocationBadge } from 'modules/Header/components';
import { Link } from 'react-router-dom';
import * as S from './styled';

type Props = {
  componentName: string;
  location: string;
};

export const Header = ({ componentName, location }: Props) => {
  return (
    <S.Header>
      <S.Nav role="navigation">
        <S.LinkWrapper>
          <Link role="link" to="/">
            Home
          </Link>
        </S.LinkWrapper>
        <S.LinkWrapper>
          <Link role="link" to="/about">
            About
          </Link>
        </S.LinkWrapper>
        <S.LinkWrapper>
          <Link role="link" to="/form">
            Form
          </Link>
        </S.LinkWrapper>
        <S.LinkWrapper>
          <Link role="link" to="/invalid">
            Invalid
          </Link>
        </S.LinkWrapper>
      </S.Nav>
      <LocationBadge componentName={componentName} location={location} />
    </S.Header>
  );
};
