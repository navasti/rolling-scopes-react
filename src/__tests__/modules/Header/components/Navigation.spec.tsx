import { Navigation } from 'modules/Header/components';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { LINKS } from 'appConstants';
import { capitalize } from 'utils';

describe('Navigation', () => {
  it('navigation component should render links and match snapshot', () => {
    const { container } = render(<Navigation />, { wrapper: BrowserRouter });
    expect(container).not.toBeEmptyDOMElement();
    LINKS.forEach((link) => expect(screen.getByText(capitalize(link))).toBeInTheDocument());
    expect(container).toMatchSnapshot();
  });
  it('navigation links should change the route', () => {
    const { container } = render(<Navigation />, { wrapper: BrowserRouter });
    const links = container.querySelectorAll('a');
    expect(links).toHaveLength(LINKS.length);
    for (let i = 0; i <= LINKS.length - 1; i++) {
      expect(links[i]).toHaveAttribute('href', `/${LINKS[i] === 'home' ? '' : LINKS[i]}`);
    }
  });
});
