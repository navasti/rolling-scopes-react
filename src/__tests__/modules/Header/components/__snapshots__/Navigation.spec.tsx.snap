// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation navigation component should render links and match snapshot 1`] = `
<div>
  <nav
    class="sc-hKMtZM jcSmlG"
  >
    <div
      class="sc-eCYdqJ ioBEYj"
    >
      <a
        href="/"
      >
        Home
      </a>
      <a
        href="/about"
      >
        About
      </a>
      <a
        href="/form"
      >
        Form
      </a>
      <a
        href="/invalid"
      >
        Invalid
      </a>
      <a
        href="/details"
      >
        Details
      </a>
    </div>
  </nav>
</div>
`;
